### PR TITLE
try to use fugit::Duraction for CountDown::Time

### PR DESCRIPTION
**Pull request by burrbull**
~~Not for SysTick for now~~ Use 1 microsecond as minimal timeout for SysTick `Counter`

This PR lets to use `fugit` units for timer including values bigger then 1 second.
Sampling of timer (time of 1 tick) now can be set in by FREQ generic constant.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -19,9 +19,12 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
 
 ### Changed
 
+- [breaking-change] use `fugit::Duraction` for `CountDown::Time` [#381]
```
Comment (dbrgn) [inline CHANGELOG.md:22]: Typo: Duraction -> Duration (also in the PR title). Maybe add "instead of a frequency" to the line?

File: CHANGELOG.md [hunk 2]
```diff
@@ -19,9 +19,12 @@ and this project adheres to [Semantic Versioning](http://semver.org/).
...
+- [breaking-change] Remove all deprecated
```
Comment (dbrgn) [inline CHANGELOG.md:23]: All deprecated what? 🙂

File: CHANGELOG.md (hunk 1 above)
Comment (burrbull) [reply CHANGELOG.md]: Thanks

File: CHANGELOG.md (hunk 2 above)
Comment (burrbull) [reply CHANGELOG.md:23]: Don't pay attention. Accidentally deleted in previous PR

**Comment by burrbull** [comment]
cc @andresv I've tried to implement `atat::Clock` also. Is this correspond to your vision?

**Comment by andresv** [comment]
@burrbull yes I think currently this is what must be done, `embedded-hal` did not have suitable traits therefore `atat::Clock` was born.

It is a little bit inconvenient to use `atat::Clock` on embedded platforms that do not use fugit based timing (almost all HALs at the moment) because essentially you have to make a new type and impl Clock for that which probably involves reading timer counter from a specific register using unsafe.

Also last time I read that `CountDown` is maybe thrown away from `embedded-hal` v1.0, so I guess we do not have anything stable for a long time.

**Comment by burrbull** [comment]
> It is a little bit inconvenient to use `atat::Clock` on embedded platforms that do not use fugit based timing (almost all HALs at the moment) because essentially you have to make a new type and impl Clock for that which probably involves reading timer counter from a specific register using unsafe.

Before port implementation to other HALs (and merge to this one), I must be sure it works as expected (I'm not sure it is so).

**Comment by andresv** [comment]
Do you mean that `fugit` itself is maybe not suitable or what do you mean by that?

**Comment by burrbull** [comment]
> Do you mean that `fugit` itself is maybe not suitable or what do you mean by that?

I mean I need someone test this and told me if I made mistake.
And yes, I still not sure about fugit.

**Comment by burrbull** [comment]
Now it just adds new structs, so no breaking changes.

**Review by therealprof — APPROVED** [review]
Very nice!

bors r+

**Pull request merged**